SportHub.IntegrationTests/AuthorizationControllerTests.cs

```diff
@@ -0,0 +1,256 @@
+using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Moq;
+using MySql.Data.MySqlClient;
+using SportHub.API;
+using SportHub.Business;
+using SportHub.Business.Implementations;
+using SportHub.Controllers;
+using SportHub.Data.Interfaces;
+using SportHub.Data.Repositories;
+using Microsoft.Extensions.Configuration;
+using SportHub.Data.DTO;
+using SportHub.Data.Entities;
+
+namespace SportHub.IntegrationTests
+{
+    [TestFixture]
+    [SingleThreaded]
+    public class AuthorizationControllerTests
+    {
+        private Mock<ILogger<AuthController>> _loggerMock;
+        private AuthController _authController;
+        private IUserService _userService;
+        private IJwtService _jwtService;
+        private IEmailService _emailService;
+        private IMapper _mapper;
+        
+        private User _testUser;
+        private UserRequestDto _testUserRequestDto;
+        private string _refreshToken;
+        private string _userId;
+        
+        [SetUp]
+        public void SetUp()
+        {
+            _loggerMock = new Mock<ILogger<AuthController>>();
+            _mapper = CreateMapper();
+            _emailService = CreateEmailService();
+            _jwtService = CreateJwtService();
+            _userService = CreateUserService();
+            
+            _authController = new AuthController(_loggerMock.Object, _userService, _jwtService, _mapper);
+            
+            // Prepare cookies response context for controller
+            var httpContext = new Mock<HttpContext>();
+            var cookies = new Mock<IResponseCookies>();
+            var response = new Mock<HttpResponse>();
+            
+            response.SetupGet(r => r.Cookies).Returns(cookies.Object);
+            
+            httpContext.SetupGet(c => c.Response).Returns(response.Object);
+
+            _authController.ControllerContext.HttpContext = httpContext.Object;
+            
+            _testUser = new User
+            {
+                FirstName = "testUserFirstName", 
+                LastName = "testUserLastName", 
+                Email = "testUserEmail@gmail.com", 
+                Password = "testUserPassword",
+            };
+            
+            _testUserRequestDto = new UserRequestDto
+            {
+                FirstName = "testUserFirstName", 
+                LastName = "testUserLastName", 
+                Email = "testUserEmail@gmail.com", 
+                Password = "testUserPassword",
+            };
+        }
+
+        [Test, Order(1)]
```

Comment:
It's bad practice to have order - remind me to discuss it on next meeting

Reply:
I need order to be sure that inserting is before activation user account and so on